### ISSUE TITLE
Syscalls marked as untested

### ISSUE THREAD

**Issue by rth** [kripken/emscripten#7194]
While investigating a sefault in `syscall320` / `utimensat` in https://github.com/iodide-project/pyodide/issues/192 , I discovered that are are a number of segfaults that are marked as TODO in,
https://github.com/kripken/emscripten/blob/ab8337c30ae1c54088e8debca170f31bd663ea19/system/lib/fetch/asmfs.cpp#L1489-L1509
(last changes 2 years ago).

At the same time the corresponding implementations appears to exist in `src/library_syscall.js` but a number are marked with,
```c
#if SYSCALL_DEBUG
    err('warning: untested syscall');
#endif
```

For instance if we take `syscall301` (`unlinkat`), we have,
1. **system/lib/fetch/asmfs.cpp**
   ```
   // TODO: syscall301: unlinkat
   ```
2. **src/library_syscall.js**
   ```js
     __syscall301: function(which, varargs) { // unlinkat
    #if SYSCALL_DEBUG
    err('warning: untested syscall');
    #endif
    // followed by an implementation
   ```
3. **tests/other/unlink/test.cpp**
   ```cpp
   if (unlinkat(AT_FDCWD, filename, 0)) {
   // [...]
   if (unlinkat(AT_FDCWD, dirname, AT_REMOVEDIR)) {
   ```
   meaning that it does appear to be tested. 

This information appears to be out of date / contradictory, or am I missing something?

In general, when building CPython in pyodide I'm wondering which syscalls should be disabled (if they are indeed untested and can lead to segfaults) and which ones should be safe to use. What would be the best way to determine this with emscripten?

**Comment by kripken** [comment]
Very possibly the "untested" comments are out of date, yes. Searching for use of the syscall in the test suite might be the best thing.

**Comment by rth** [comment]
Thanks for the response! Indeed, after a more detailed analysis from the above list only `unlinkat` is marked as "untested" while it is tested. The rest of comments are still valid. Will make a PR to remove the outdated comment.

**Comment by rth** [comment]
Comments consistency fixed in #7240
